component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,404 @@
+     */
+    private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientId, String code, String clientSecret,
+                                                String callbackurl) throws AuthenticationFailedException {
+
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint).setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setClientId(clientId).setClientSecret(clientSecret).setRedirectURI(callbackurl).setCode(code)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException("Communication errors occurred between the OAuth Recourse Server " +
+                    "and the OAuth Authorization Server when all the parameters[client_Id : " + clientId +
+                    " ,client_Secret, token_End_Point: " + tokenEndPoint + " and callbackurl: " + callbackurl +
+                    "] are built a Form-Encoded Body Parameter." , e);
+        }
+        return accessRequest;
+    }
+
+    /**
+     * Get the user info from user info endpoint.
+     *
+     * @param url         User info endpoint.
+     * @param accessToken Access token.
+     * @return Response string.
+     */
+    private String getUserInfoJson(String url, String accessToken) throws AuthenticationFailedException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("User info endpoint: " + url);
+        }
+        BufferedReader reader = null;
+        StringBuilder result = null;
+        Integer statusCode;
+        String line;
+        HttpGet httpget = null;
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            httpget = new HttpGet(url);
+            httpget.setHeader(NuxeoAuthenticatorConstants.AUTHORIZATION,
+                    NuxeoAuthenticatorConstants.AUTHENTICATION_BEARER + accessToken);
+            result = new StringBuilder();
+            statusCode=httpclient.execute(httpget).getStatusLine().getStatusCode();
+            reader = new BufferedReader(new InputStreamReader(httpclient.execute(httpget).getEntity().getContent()));
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+            if (statusCode == HttpStatus.SC_OK) {
+                if (log.isDebugEnabled()) {
+                    log.debug("logged in User info: " + result.toString());
+                }
+                return result.toString();
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Couldn't get the successful response while getting the " +
+                            "UserInfo for the logged in User. Status code: " + statusCode + " and Response: " +
+                            result.toString());
+                }
+                throw new AuthenticationFailedException(" Couldn't get the successful response while getting the " +
```

Comment:
Let's change this to something like, 'Unable to get a successful response ......'